grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -1028,6 +1028,20 @@ class ProductController {
         render(template: 'productSynonyms', model: [productInstance: product, inputValues: inputValues])
     }
 
+    /**
+     * Edit existing product synonym
+     */
+    def editProductSynonym = {
+        println "editProductSynonym() " + params
+        productService.editProductSynonym(params['synonym.id'], params.synonymTypeCode, params.synonym, params.locale)
+        render (status: 200, text: "successfully edited synonym")
+    }
+
+    def editProductSynonymDialog = {
+        Synonym synonym = Synonym.read(params.id)
+        render(template: 'productSynonymsEdit', model: [product: synonym.product, synonym: synonym])
```

Comment:
I know this comment and below might be nitpicky - but just in case shouldn't we add as much null checks as we can? (here, `synonym?.product` - I don't know, in case someone deleted the product "in the mean time")

Reply:
It wouldn't hurt (in case someone removes the checks above it), but it's unnecessary in the current context. But again, it doesn't hurt and I like to be defensive assuming it's not decreasing readability.